File: Habitica/src/com/habitrpg/android/habitica/ui/fragments/setup/IntroFragment.java
package com.habitrpg.android.habitica.ui.fragments.setup;

import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.habitrpg.android.habitica.R;
import com.habitrpg.android.habitica.ui.fragments.BaseFragment;

import butterknife.Bind;
import butterknife.ButterKnife;

public class IntroFragment extends BaseFragment {

    @Bind(R.id.titleTextView)
    TextView titleTextView;

    @Bind(R.id.descriptionTextView)
    TextView descriptionTextView;

    @Bind(R.id.imageView)
    ImageView imageView;

    Drawable image;
    String title;
    String description;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        View v = inflater.inflate(R.layout.fragment_intro, container, false);

        ButterKnife.bind(this, v);

        if (this.image != null) {
            this.imageView.setImageDrawable(this.image);
        }

        if (this.title != null) {
            this.titleTextView.setText(this.title);
        }

        if (this.description != null) {
            this.descriptionTextView.setText(this.description);
        }

        return v;
    }

    public void setImage(Drawable image) {
        this.image = image;
        if (this.imageView != null && image != null) {
            this.imageView.setImageDrawable(image);
        }
    }

    public void setTitle(String text) {
        this.title = text;
        if (this.titleTextView != null && text != null) {
            this.titleTextView.setText(text);
        }
    }

    public void setDescription(String text) {
        this.description = text;
        if (this.descriptionTextView != null && text != null) {
            this.descriptionTextView.setText(text);
        }
    }

}

File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/fragments/setup/IntroFragment.kt
package com.habitrpg.android.habitica.ui.fragments.setup

import android.graphics.drawable.Drawable
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.habitrpg.android.habitica.components.UserComponent
import com.habitrpg.android.habitica.databinding.FragmentIntroBinding
import com.habitrpg.android.habitica.ui.fragments.BaseFragment
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class IntroFragment : BaseFragment<FragmentIntroBinding>() {

    override var binding: FragmentIntroBinding? = null

    override fun createBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentIntroBinding {
        return FragmentIntroBinding.inflate(inflater, container, false)
    }

    private var image: Drawable? = null
    private var titleImage: Drawable? = null
    private var subtitle: String? = null
    private var title: String? = null
    private var description: String? = null
    private var backgroundColor: Int? = null

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        if (this.image != null) {
            binding?.imageView?.setImageDrawable(this.image)
        }

        if (this.titleImage != null) {
            binding?.titleImageView?.setImageDrawable(this.titleImage)
        }

        if (this.subtitle != null) {
            binding?.subtitleTextView?.text = this.subtitle
        }

        if (this.title != null) {
            binding?.titleTextView?.text = this.title
        }

        if (this.description != null) {
            binding?.descriptionTextView?.text = this.description
        }

        backgroundColor?.let {
            binding?.containerView?.setBackgroundColor(it)
        }
    }


    fun setImage(image: Drawable?) {
        this.image = image
        if (image != null) {
            binding?.imageView?.setImageDrawable(image)
        }
    }

    fun setTitleImage(image: Drawable?) {
        this.titleImage = image
        binding?.titleImageView?.setImageDrawable(image)
    }

    fun setSubtitle(text: String?) {
        this.subtitle = text
        binding?.subtitleTextView?.text = text
    }

    fun setTitle(text: String?) {
        this.title = text
        binding?.titleTextView?.text = text
    }

    fun setDescription(text: String?) {
        this.description = text
        binding?.descriptionTextView?.text = text
    }

    fun setBackgroundColor(color: Int) {
        this.backgroundColor = color
        binding?.containerView?.setBackgroundColor(color)
    }
}
